cmd/osbuild-jobsite-builder/main.go

```diff
@@ -119,45 +118,42 @@ func (builder *Builder) GetState() State {
 	return builder.State
 }
 
-func (builder *Builder) GuardState(stateWanted State) {
+func (builder *Builder) GuardState(stateWanted State) error {
 	if stateCurrent := builder.GetState(); stateWanted != stateCurrent {
-		logrus.Fatalf("Builder.GuardState: Requested guard for %d but we're in %d. Exit", stateWanted, stateCurrent)
+		return fmt.Errorf("Builder.GuardState: Requested guard for %d but we're in %d. Exit", stateWanted, stateCurrent)
 	}
+
+	return nil
 }
 
-func (builder *Builder) RegisterHandler(h Handler) http.HandlerFunc {
+func (builder *Builder) RegisterHandler(s State, m string, h Handler) http.HandlerFunc {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if err := builder.GuardState(s); err != nil {
+			w.WriteHeader(http.StatusConflict)
+			logrus.Fatal("Received request in invalid state. Exiting")
+		}
+
+		if r.Method != m {
+			w.WriteHeader(http.StatusBadRequest)
+			logrus.Fatal("Received invalid request method. Exiting")
+		}
+
 		if err := h(w, r); err != nil {
```

Comment:
Might be useful to also log here in what state the error happened, e.g. `logrus.Fatalf("error in %v: %v", s, err)`

Reply:
Done.

Reply:
Is a push missing here maybe? The comment says "done" but the code is still the same?